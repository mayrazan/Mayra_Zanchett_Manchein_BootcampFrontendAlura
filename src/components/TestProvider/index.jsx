import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../theme';
import GlobalStyle from '../../theme/GlobalStyle';

const TestProvider = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

TestProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TestProvider;