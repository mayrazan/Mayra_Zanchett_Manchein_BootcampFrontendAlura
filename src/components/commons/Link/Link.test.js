import React from 'react';
import { render, screen } from '../../TestProvider/testUtils';
import Link from '.';

describe('<Link />', () => {
  test('renders link', () => {
    render(
      <Link href="/aboutme">
        Sobre mim
      </Link>,
    );
    const textLink = screen.getByText(/sobre mim/i);
    expect(textLink).toBeInTheDocument();
    expect(textLink).toMatchSnapshot();
  });
});
