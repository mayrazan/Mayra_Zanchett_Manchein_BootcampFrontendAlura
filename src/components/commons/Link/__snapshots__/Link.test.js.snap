// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders link 1`] = `
.c0 {
  color: inherit;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: 1;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
}

.c0:hover,
.c0:focus {
  opacity: 0.5;
}

<a
  class="c0"
  href="/aboutme"
  rel="noreferrer"
>
  Sobre mim
</a>
`;
